storage/src/main/java/org/apache/uniffle/storage/handler/impl/MemoryClientReadHandler.java

```diff
@@ -66,18 +76,32 @@ public MemoryClientReadHandler(
       int readBufferSize,
       ShuffleServerClient shuffleServerClient,
       Roaring64NavigableMap expectTaskIds,
-      boolean expectedTaskIdsBitmapFilterEnable) {
+      Roaring64NavigableMap expectBlockIds,
+      Roaring64NavigableMap processBlockIds,
+      BlockSkipStrategy blockSkipStrategy,
+      int maxBlockIdRangeSegments) {
     this.appId = appId;
     this.shuffleId = shuffleId;
     this.partitionId = partitionId;
     this.readBufferSize = readBufferSize;
     this.shuffleServerClient = shuffleServerClient;
     this.expectTaskIds = expectTaskIds;
-    this.expectedTaskIdsBitmapFilterEnable = expectedTaskIdsBitmapFilterEnable;
+    this.expectBlockIds = expectBlockIds;
+    this.processBlockIds = processBlockIds;
+    this.blockSkipStrategy = blockSkipStrategy;
+    this.maxBlockIdRangeSegments = maxBlockIdRangeSegments;
   }
 
   @Override
   public ShuffleDataResult readShuffleData() {
+    if (BlockSkipStrategy.BLOCKID_RANGE.equals(blockSkipStrategy) && lastBlockId == Constants.INVALID_BLOCK_ID) {
```

Comment:
Why do we judge the `lastBlockId == Constants.INVALID_BLOCK_ID`?

Reply:
We only need to build the blockId range at the first time the handler read.

Reply:
OK, got it.